src/videojs-hls.js

```diff
@@ -1008,17 +999,25 @@ videojs.HlsHandler.prototype.loadSegment = function(segmentInfo) {
     // decrease in network performance or a server issue.
     timeout: (segment.duration * 1.5) * 1000
   }, function(error, request) {
-    // the segment request is no longer outstanding
-    self.segmentXhr_ = null;
+    // This is a timeout of a previously aborted segment request
+    // so simply ignore it
+    if (!self.segmentXhr_ || request !== self.segmentXhr_) {
+      return;
+    }
 
     // if a segment request times out, we may have better luck with another playlist
     if (request.timedout) {
+      // the segment request is no longer outstanding
+      self.segmentXhr_ = null;
```

Comment:
It looks like we're still nulling `segmentXhr_` for all code paths after this point. Why repeat it within each of the conditionals?